dev/tools/shared/utils.py

```
@@ -55,6 +55,67 @@ def git_sha():
+    Otherwise:
+    - If the repository is clean, returns the HEAD commit date.
+    - If the repository is dirty, returns the mtime of the most recently modified file.
+    """
+    env_date = os.getenv("BUILD_DATE")
+    if env_date:
+        return env_date
+
+    repo_root = get_repo_root()
+
+    res = subprocess.run(
+        ["git", "status", "--porcelain", "-z"],
```

Ruff (0.16.2)

[error] Starting a process with a partial executable path

(S607)